src/trace/dma-trace.c

```diff
@@ -260,28 +293,63 @@ static int dma_trace_buffer_init(struct dma_trace_data *d)
 
 	k_spin_unlock(&d->lock, key);
 
-	return 0;
-}
+#ifdef __ZEPHYR__
+#define ZEPHYR_VER_OPT " zephyr:" META_QUOTE(BUILD_VERSION)
+#else
+#define ZEPHYR_VER_OPT
+#endif
 
-static void dma_trace_buffer_free(struct dma_trace_data *d)
-{
-	struct dma_trace_buf *buffer = &d->dmatb;
-	k_spinlock_key_t key;
+	/* META_QUOTE(SOF_SRC_HASH) is part of the format string so it
+	 * goes to the .ldc file and does not go to the firmware
+	 * binary. It will be different from SOF_SRC_HASH in case of
+	 * mismatch.
+	 */
+#define SOF_BANNER_COMMON  \
+	"FW ABI 0x%x DBG ABI 0x%x tags SOF:" SOF_GIT_TAG  ZEPHYR_VER_OPT  \
+	" src hash 0x%08x (ldc hash " META_QUOTE(SOF_SRC_HASH) ")"
```

Comment:
Can you please move these macros down, back together with the only place they are used?